src/messages.rs

```diff
@@ -489,7 +489,10 @@ pub enum MessageContent {
         message_id: MessageId,
     },
     /// Sent to all connected peers when our own group splits
-    GroupSplit(Prefix<XorName>),
+    GroupSplit {
+        prefix: Prefix<XorName>,
+        joining_node: XorName,
```

Comment:
Actually we're still using the unaltered joining node name as the source authority, right? Then we don't need this field and can just use the source address?